### ISSUE TITLE
Help for the model

### ISSUE THREAD

**Issue by Fyzjym** [MattAlexMiracle/SmartPatch#1]
Sir, the method proposed in your article has effectively improved Ganwriting.
I have a question about evaluating Fid. I am curious how Ganwriting got 120~130 fid. Can you tell me how do you calculate fid specifically?
Would you like to share the samrtpatch(NaivePatch)pre-trained model.
I am sorry for my recklessness.
good luck.

**Comment by MattAlexMiracle** [comment]
Hi,
I uploaded some trained models here:
https://faubox.rrze.uni-erlangen.de/getlink/fiAjmuhP9gfZYdCQHBaBoy2X/
Make sure to download them soon, as I don't know how long these are going to be available on the server.
The FID is calculated between
The IAM dataset and the synthetic-IAM dataset, as described in the paper:
i.e. For each element in the RWTH-Aachen split, we generate an image with the same writer and content, such that we get a new synthetic dataset with the same structure as the original.
(see generatePseudoIAM.py, this assumes the IAM dataset is in "dataset/", the RWTH-split is in "groundtruth/" and the target for the synthetic dataset is given as an argument)
We then compare
The distribution of the synthetic dataset with the real one via FID. (see: "fid_folder")

The synthetic IAM approach is necessary, since if you just generate random text with random style, the distribution will be different:
We want 
style~WordStyles
and not, for example
style~N(0,I)
So simply using random noise as the writer-style would sample out-of-distribution and therefore no longer be comparable to IAM via FID.

Also make sure to train GANwriting and their derivatives long enough: The settings in main with a 500 early-stopping threshhold should work fine, but pushing that too low will not give you that great of an result. (the loss curve tend to be very deceiving, as everything, including the loss-networks, are trained jointly)

Further, keep in mind that the names of the techniques in this repo are different from the ones in the paper (see: README)

**Comment by Fyzjym** [comment]
Sir, thank you for providing such a patient and professional answer.
After reading your answer, I have realized that there is a problem in my calculation process.
I am very respectful of your move to make the pre-training model public!
Finally, good luck.
